### PR TITLE
Pass references

### PR DESCRIPTION
**Pull request by sjanssen2**
1) Added optional inputs for reference alignment and reference phylogeny.
2) using small reference for unit testing
3) added validation of reference, i.e. do all alignment row IDs occur as tips in the phylogeny
4) post processing of insertion tree to ensure all nodes have an explicit branch length (needed by downstream q2 plugins)

### CODE REVIEW

**Comment by sjanssen2** [comment]
@wasade can you please take a look

**Comment by sjanssen2** [comment]
want to take a look @tkosciol

File: q2_fragment_insertion/plugin_setup.py [hunk 1]
```diff
@@ -48,10 +48,20 @@
...
-    input_descriptions={'representative_sequences': "The sequences to insert"},
+    input_descriptions={
+        'representative_sequences': "The sequences to insert",
+        'reference_alignment':
+        ('The reference multiple nucleotide alignment used '
+         'to construct the reference phylogeny.'),
+        'reference_phylogeny':
+        ('The rooted reference phylogeny. Must be in sync '
+         'with reference-alignment, i.e. each tip name must'
+         ' have exactly one corresponding row.')},
```
Comment (wasade) [inline q2_fragment_insertion/plugin_setup.py]: row? or record?

Comment (sjanssen2) [reply q2_fragment_insertion/plugin_setup.py]: hm, not sure if I understand the semantic difference between alignment row and record? Do you mean one sequence within the alignment with a "record"?

Comment (wasade) [reply q2_fragment_insertion/plugin_setup.py]: Row is an unusual way to describe a sequence

Comment (sjanssen2) [reply q2_fragment_insertion/plugin_setup.py]: its not really a sequence, since it contains gaps

**Comment by sjanssen2** [comment]
@mortonjt should you have some time, could you take a look at this PR since you have some experience writing q2 plugins?

**Comment by wasade** [comment]
I'm :+1:

**Pull request merged**